client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,476 @@
+            set_message_for_confirmed_transactions(
+                progress_bar,
+                context
+                    .num_confirmed_transactions
+                    .load(std::sync::atomic::Ordering::Relaxed),
+                context.total_transactions,
+                None,
+                blockhashdata.last_valid_blockheight,
+                &format!(
+                    "Sending {}/{} transactions",
+                    counter + 1,
+                    current_transaction_count,
+                ),
+            );
+        }
+    }
+    Ok(())
+}
+
+async fn confirm_transactions_till_block_height_and_resend_unexpired_transaction_over_tpu(
+    progress_bar: &Option<indicatif::ProgressBar>,
+    tpu_client: &Option<QuicTpuClient>,
+    context: &SendingContext,
+) {
+    let unconfirmed_transasction_map = context.unconfirmed_transasction_map.clone();
+    let num_confirmed_transactions = context.num_confirmed_transactions.clone();
+    let current_blockheight = context.current_blockheight.clone();
+    let total_transactions = context.total_transactions;
+
+    let transactions_to_confirm = unconfirmed_transasction_map.len();
+    let max_valid_block_height = unconfirmed_transasction_map
+        .iter()
+        .map(|x| x.last_valid_blockheight)
+        .max();
+
+    if let Some(mut max_valid_block_height) = max_valid_block_height {
+        if let Some(progress_bar) = progress_bar {
+            set_message_for_confirmed_transactions(
+                progress_bar,
+                num_confirmed_transactions.load(std::sync::atomic::Ordering::Relaxed),
+                total_transactions,
+                Some(current_blockheight.load(Ordering::Relaxed)),
+                max_valid_block_height,
+                &format!(
+                    "Waiting for next block, {transactions_to_confirm} transactions pending..."
+                ),
+            );
+        }
+
+        // wait till all transactions are confirmed or we have surpassed max processing age for the last sent transaction
+        while !unconfirmed_transasction_map.is_empty()
+            && current_blockheight.load(Ordering::Relaxed) < max_valid_block_height
+        {
+            let blockheight = current_blockheight.load(Ordering::Relaxed);
+
+            if let Some(progress_bar) = progress_bar {
+                set_message_for_confirmed_transactions(
+                    progress_bar,
+                    num_confirmed_transactions.load(std::sync::atomic::Ordering::Relaxed),
+                    total_transactions,
+                    Some(blockheight),
+                    max_valid_block_height,
+                    "Checking transaction status...",
+                );
+            }
+
+            if let Some(tpu_client) = tpu_client {
+                let instant = Instant::now();
+                // retry sending transaction only over TPU port / any transactions sent over RPC will be automatically rebroadcast by the RPC server
```

Comment:
Long line > 120 symbols

```suggestion
                // retry sending transaction only over TPU port
                // any transactions sent over RPC will be automatically rebroadcast by the RPC server
```